uwrt_utils/src/uwrt_can.cpp

```diff
@@ -0,0 +1,156 @@
+#include "uwrt_utils/uwrt_can.h"
+
+#include <ros/ros.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
+#include <cerrno>
+#include <cstring>
+
+namespace uwrt_utils {
+
+// static constexpr need to be declared again in cpp file (another good one c++ 11)
```

Comment:
fixed in c++ 17. maybe leave a comment to remove once ros upgrades its default c++ version. You could also just set the compile options to have this file compile in c++17

Reply:
Idk if theres a way to detect if T is a primitive type or a snob struct